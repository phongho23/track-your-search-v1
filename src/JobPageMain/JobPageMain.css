.JobPageMain {
    height: 100%;
    padding: 10px;
    width: 100%;
    margin: auto;
    align-content: center;
    align-items: center;
  }
  
  .JobPageMain__jobtitle {
    margin: auto;
    align-content: center;
    justify-content: center;
  }
  
  .JobPageMain__companyname {
    margin: auto;
    align-content: center;
    justify-content: center;
  }
  
  .JobPageMain__interview {
    margin: auto;
    align-content: center;
    justify-content: center;
  }
  
  .JobPageMain__postedurl {
    margin: auto;
    align-content: center;
    justify-content: center;
  }
  
  .JobPageMain__jobrating {
    margin: auto;
    align-content: center;
    justify-content: center;
  }
  
  .JobPageMain__content {
    margin: auto;
    align-content: center;
    justify-content: center;
  }
  
  .jobInfo {
    margin: auto;
    align-content: center;
    align-items: center;
    background-color: #8F2222;
    width: 70%;
    border-radius: 10px;
  }
  
  .JobPageMain__content {
    color: white;
    max-width: 380px;
  }
  
  .JobPageMain__jobtitle {
    color: white;
    max-width: 380px;
    padding-top: 12px;
  }
  
  .JobPageMain__companyname {
    color: white;
    max-width: 380px;
    padding-top: 12px;
  }
  
  .JobPageMain__interview {
    color: white;
    max-width: 380px;
    padding-top: 12px;
  }
  
  .JobPageMain__jobrating {
    color: white;
    max-width: 380px;
    padding-top: 12px;
  }
  
  .JobPageMain__postedurl {
    color: white;
    max-width: 380px;
    padding-top: 12px;
  }
  