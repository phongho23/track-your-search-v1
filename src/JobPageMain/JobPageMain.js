import React from 'react'
import Job from '../Job/Job'
import ApiContext from '../ApiContext'
import { findJob } from '../jobs-helpers'
import './JobPageMain.css'

export default class JobPageMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }

  static contextType = ApiContext

  handleDeleteJob = jobId => {
    this.props.history.push(`/`)
  }

  render() {

    const { jobs=[] } = this.context
    const { jobId } = this.props.match.params
    const job = findJob(jobs, jobId) || { content: '' }

  return (
    <section className='JobPageMain'>
      <Job
        id={job.id}
        jobtitle={job.jobtitle}
        modified={job.modified}
        onDeleteJob={this.handleDeleteJob}
      />
      <div className='jobInfo'>
      <div className='JobPageMain__jobtitle'>
        <b><p>Job Title:</p></b>
        {job.jobtitle}
      </div>

      <div className='JobPageMain__companyname'>
        <b><p>Company Name:</p></b>
        {job.companyname}
      </div>

      <div className='JobPageMain__postedurl'>
        <b><p>Job Listing URL/Info:</p></b>
        {job.postedurl}
      </div>

      <div className='JobPageMain__interview'>
        <b><p>Interview Details:</p></b>
        {job.interview}
      </div>

      <div className='JobPageMain__jobrating'>
        <b><p>Opportunity Rating:</p></b>
        {job.jobrating}
      </div>

      <div className='JobPageMain__content'>
        <b><p>Job Description:</p></b>
        {job.content.split(/\n \r|\n/).map((para, i) =>
          <p key={i}>{para}</p>
        )}
      </div>
      </div>
    </section>
    )
  }
}
