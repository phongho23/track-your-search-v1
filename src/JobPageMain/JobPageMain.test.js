import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import JobPageMain from './JobPageMain'

describe(`JobPageMain component`, () => {
  it('renders a .JobPageMain by default', () => {
    const wrapper = shallow(<JobPageMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it.skip('renders a Job with job prop', () => {
    const props = {
      match: {
        params: {
          jobId: 1
        }
      }
    }
    const context = {
      jobs: [{
        id: `1`,
        name: `Software Engineer I`, 
        modified: `2021-01-03T00:00:00.000Z`, 
        weekId: `1`, 
        jobtitle: `Software Engineer III`,
        companyname: "Target",
        postedurl: `http://www.URLholder.com/`,
        interview: `Scheduled for 05/20/2021`,
        jobrating: `5`,
        content: "Corporis accusamus placeat quas non voluptas. Harum fugit molestias qui. Velit ex animi reiciendis quasi."
      }]
    }
    const job = shallow(<JobPageMain {...props} />, context)
      .find('Job')
    expect(toJson(job)).toMatchSnapshot()
  })

  // enzyme doesn't yet support React.createContext
  it.skip(`splits the content by \\n or \\n\\r, with a p foreach`, () => {
    const props = {
      match: {
        params: {
          jobId: '1'
        }
      }
    }

    const jobsContextWithDifferentContent = [
      {
        jobs: [
          {
            id: `1`,
            content: "Content with n r.\n \rafter n r.",
          }
        ]
      },
      {
        jobs: [
          {
            id: `1`,
            content: "Content with n.\nafter."
          }
        ]
      }
    ]

    jobsContextWithDifferentContent.forEach(context => {
      const content = shallow(<JobPageMain {...props} />, context)
        .find('JobPageMain__content')
      expect(toJson(content)).toMatchSnapshot()
    })
  })
})
