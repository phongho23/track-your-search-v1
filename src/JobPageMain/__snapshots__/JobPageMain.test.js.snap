// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobPageMain component renders a .JobPageMain by default 1`] = `
<section
  className="JobPageMain"
>
  <Job
    onDeleteJob={[Function]}
  />
  <div
    className="jobInfo"
  >
    <div
      className="JobPageMain__jobtitle"
    >
      <b>
        <p>
          Job Title:
        </p>
      </b>
    </div>
    <div
      className="JobPageMain__companyname"
    >
      <b>
        <p>
          Company Name:
        </p>
      </b>
    </div>
    <div
      className="JobPageMain__postedurl"
    >
      <b>
        <p>
          Job Listing URL/Info:
        </p>
      </b>
    </div>
    <div
      className="JobPageMain__interview"
    >
      <b>
        <p>
          Interview Details:
        </p>
      </b>
    </div>
    <div
      className="JobPageMain__jobrating"
    >
      <b>
        <p>
          Opportunity Rating:
        </p>
      </b>
    </div>
    <div
      className="JobPageMain__content"
    >
      <b>
        <p>
          Job Description:
        </p>
      </b>
      <p
        key="0"
      />
    </div>
  </div>
</section>
`;
